clients/pkg/promtail/client/batch_test.go

```diff
@@ -164,3 +164,17 @@ func TestHashCollisions(t *testing.T) {
 		assert.Equal(t, ls1.String(), req.Streams[1].Labels)
 	}
 }
+
+func BenchmarkLabelsMapToString(b *testing.B) {
+	labelSet := make(model.LabelSet)
+	labelSet["label"] = "value"
+	labelSet["label1"] = "value"
+	labelSet["label2"] = "value"
+	labelSet["__tenant_id__"] = "value"
+
+	ReservedLabelTenantID := "__tenant_id__"
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		_ = labelsMapToString(labelSet, model.LabelName(ReservedLabelTenantID))
```

Comment:
I'm bit concerned about using `_` to store the result here. Mainly because of some internal compiler optimizations. Compiler may think why call this func `labelsMapToString` in tight loop if result is not used at all.

I would prefer to have some variable outside loop to store the result.

```
var result string
for i:=0; i<b.N; i++ {
    result = ....
}
```

and compare the results again.

Reply:
Thanks, I re-ran the benchmark with the code below and get the same results :relieved: 

```
// store the result to a package level variable
// so the compiler cannot eliminate the Benchmark itself.
var result string

func BenchmarkLabelsMapToString(b *testing.B) {
	labelSet := make(model.LabelSet)
	labelSet["label"] = "value"
	labelSet["label1"] = "value"
	labelSet["label2"] = "value"
	labelSet["__tenant_id__"] = "value"

	ReservedLabelTenantID := "__tenant_id__"
	b.ResetTimer()
	var r string
	for i := 0; i < b.N; i++ {
		// store in r prevent the compiler eliminating the function call.
		r = labelsMapToString(labelSet, model.LabelName(ReservedLabelTenantID))
	}
	result = r
}
```